src/test/daml/Daml/Finance/Instrument/Swap/Test/Currency.daml

```diff
@@ -0,0 +1,97 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Swap.Test.Currency where
+
+import DA.Date
+import DA.Map qualified as M (fromList)
+import DA.Set (singleton)
+import Daml.Finance.Data.Reference.HolidayCalendar
+import Daml.Finance.Holding.Fungible qualified as Fungible (Factory(..))
+import Daml.Finance.Instrument.Swap.Test.Util
+import Daml.Finance.Interface.Types.Date.Calendar
+import Daml.Finance.Interface.Types.Date.DayCount
+import Daml.Finance.Interface.Types.Date.RollConvention
+import Daml.Finance.Test.Util.Account qualified as Account (createAccount, createFactory, credit)
+import Daml.Finance.Test.Util.Common (createParties)
+import Daml.Finance.Test.Util.Instrument qualified as Instrument (originate)
+import Daml.Script
+
+-- Calculate fix rate payments on a currency swap, including lifecycling and creation of new instrument version.
+run : Script ()
+run = script do
+  [custodian, issuer, investor, calendarDataProvider, publicParty] <-
+    createParties ["Custodian", "Issuer", "Investor", "Calendar Data Provider", "PublicParty"]
+
+  -- Account and holding factory
+  let pp = [("FactoryProvider", singleton publicParty)]
+  accountFactoryCid <- toInterfaceContractId <$> Account.createFactory custodian pp
+  holdingFactoryCid <- toInterfaceContractId <$> submitMulti [custodian] [] do
+    createCmd Fungible.Factory with provider = custodian; observers = M.fromList pp
+
+  -- Create accounts
+  [custodianAccount, investorAccount] <- mapA (Account.createAccount "Default Account" [publicParty] accountFactoryCid holdingFactoryCid [] custodian) [custodian, investor]
+
+  -- Distribute commercial-bank cash
+  now <- getTime
+  let observers = [("PublicParty", singleton publicParty)]
+  cashInstrumentCid <- Instrument.originate custodian issuer "USD" "US Dollars" observers now
+  foreignCashInstrumentCid <- Instrument.originate custodian issuer "EUR" "Euro" observers now
+
+  -- Create and distribute swap
+  -- Fix vs fix currency swap: 3% p.a. in USD vs 2% p.a. in EUR, payment every 3M
+  let
+    issueDate = date 2019 Jan 16
+    firstPaymentDate = date 2019 Feb 15
+    maturityDate = date 2019 May 15
+    issuerPaysBase = False
+    baseRate = 0.03
+    foreignRate = 0.02
+    fxRate = 1.1
+    paymentPeriod = M
+    paymentPeriodMultiplier = 3
+    dayCountConvention = Act360
+    businessDayConvention = ModifiedFollowing
+    firstBasePaymentAmount : Decimal = 2500.0
+    firstForeignPaymentAmount : Decimal = 1833.3333
+    secondBasePaymentAmount : Decimal = 7416.6667
+    secondForeignPaymentAmount : Decimal = 5438.8889
```

Comment:
do we need to declare this as a `: Decimal`, maybe you could remove it?

Reply:
This is not needed anymore, so I have removed it. The only reason I include it is because the compiler sometimes complains during development otherwise, when I comment out parts of the code.